### PR TITLE
Address S2955: Comparing unconstrained generic values against null

### LINKED ISSUE #933: Address S2955: Generic parameters not constrained to reference types should not be compared to "null"
Address existing violations of [S2955: Generic parameters not constrained to reference types should not be compared to "null"](https://rules.sonarsource.com/csharp/RSPEC-2955) in the codebase and set severity to `Warning` in `Steeltoe.Debug.ruleset` and `Steeltoe.Release.ruleset`.

To find existing violations, enable the rule (see above) and rebuild `src/Steeltoe.All.sln` to make them appear in the Output window.

To address the violations, choose from the following on a case-by-case basis:
- Fix the violation by changing the code to not violate the rule
- Suppress the violation in code using `#pragma warning disable/restore`, preceded by a justification comment if not obvious

### PR DESCRIPTION
**Pull request by bart-vmware**
## Description

For some cases, I was able to add a `class` constraint to the generic type parameter to make the warning go away.
For the remaining ones, it resulted in errors, so for now, I've used a hack (`is null` instead of `== null`), which Sonar does not detect. Please advise on a case-by-case basis whether it should a) compare against `default` or b) keep checking against `null` or c) remove the check altogether.

Fixes #933

## Quality checklist
- [x] Your code complies with our [Coding Style](https://github.com/SteeltoeOSS/.github/blob/main/contributing-docs/contributing-code-style.md).
- [x] You've updated unit and/or integration tests for your change, where applicable.
- [ ] You've updated documentation for your change, where applicable.
      If your change affects other repositories, such as [Documentation](https://github.com/SteeltoeOSS/Documentation), [Samples](https://github.com/SteeltoeOSS/Samples) and/or [MainSite](https://github.com/SteeltoeOSS/MainSite), add linked PRs here.
- [x] There's an open issue for the PR that you are making. If you'd like to propose a new feature or change, please open an issue to discuss the change or find an existing issue.
- [ ] You've added required license files and/or file headers (explaining where the code came from with proper attribution), where code is copied from StackOverflow, a blog, or OSS.

### CODE REVIEW

File: src/CircuitBreaker/src/Hystrix/Collapser/RequestBatch.cs [hunk 1]
```diff
@@ -74,7 +74,7 @@ public int Size
                     collapsedRequest.CompletionSource = tcs;
 
                     CollapsedRequest<TRequestResponse, TRequestArgument> existing =
-                        arg == null ? GetOrAddNullArg(collapsedRequest) : _argumentMap.GetOrAdd(arg, collapsedRequest);
+                        arg is null ? GetOrAddNullArg(collapsedRequest) : _argumentMap.GetOrAdd(arg, collapsedRequest);
```
Comment (bart-vmware) [inline src/CircuitBreaker/src/Hystrix/Collapser/RequestBatch.cs:77]: @dtillman `arg` is of unconstrained generic type `TRequestArgument`, so comparing to `null` doesn't make sense when it's a value type (resulting in S2955 warning). How do you think this should be solved? I've tried adding a `class` constraint, but that results in errors.

File: src/Common/src/Common.Expression/Internal/Spring/Ast/Indexer.cs [hunk 2]
```diff
@@ -286,7 +286,7 @@ private T ConvertValue<T>(ITypeConverter converter, object value)
         Type targetType = typeof(T);
         var result = (T)converter.ConvertValue(value, value == null ? typeof(object) : value.GetType(), targetType);
 
-        if (result == null)
+        if (result is null)
```
Comment (bart-vmware) [inline src/Common/src/Common.Expression/Internal/Spring/Ast/Indexer.cs:289]: @dtillman Same question here. Allowing value types seems intentional.

File: src/Messaging/src/RabbitMQ/Core/RabbitTemplate.cs [hunk 3]
```diff
@@ -2266,7 +2266,7 @@ private void LogReceived(IMessage message)
 
         TReply reply = receiveAndReplyCallback(messageAsTReceive);
 
-        if (reply != null)
+        if (reply is not null)
```
Comment (bart-vmware) [inline src/Messaging/src/RabbitMQ/Core/RabbitTemplate.cs:2269]: @hananiel `reply` is of unconstrained generic type `TReply`, so comparing to null doesn't make sense when it's a value type (resulting in S2955 warning). How do you think this should be solved? I've tried adding a class constraint, but that results in errors.

Comment (bart-vmware) [reply src/Messaging/src/RabbitMQ/Core/RabbitTemplate.cs:2269]: @dtillman

**Review by dtillman — APPROVED** [review]

**Pull request merged**